airflow/serialization/serialized_objects.py

```diff
@@ -300,7 +319,18 @@ def _deserialize(cls, encoded_var: Any) -> Any:  # pylint: disable=too-many-retu
             raise TypeError(f'Invalid type {type_!s} in deserialization.')
 
     _deserialize_datetime = pendulum.from_timestamp
-    _deserialize_timezone = pendulum.tz.timezone
+
+    @classmethod
+    def _deserialize_timezone(cls, tz_name: str) -> Timezone:
+        try:
+            return pendulum.tz.timezone(tz_name)
+        except InvalidTimezone:
+            pass
+        try:
+            return pendulum.parse("2021-01-01T12:00:00" + tz_name).tzinfo
```

Comment:
This looks more than slightly odd -- when is this case hit?

Reply:
From any of the new cases provided in the tests_dag_serialization.py:test_deserialization_start_date()

But as an example if the DAG start_date is `pendulum.parse("2019-08-01T00:00:00.000+01:00"` then DAG timezone will be a `pendulum.tz.timezone.FixedOffset` with name `+01:00` so that it's what it will be serialized to. 

Now in the deserialization step `pendulum.timezone('+01:00')` will raise a `InvalidTimezone` because that only work for "named" timezones (and not all I must add). See https://github.com/sdispater/pendulum/issues/554 

So if we want to parse `+01:00` back to a timezone object, `pendulum.timezone()` won't do it, we need to "hack it" by creating a "fake" iso 8601 string with the `+01:00` and using `pendulum.parse()` to construct a `FixedTimezone()` object for us. 

I couldn't find any other method in pendulum's to convert the string `'+01:00` to a FixedTimezone/Timezone/`datetime.tzinfo`